x/chainlink/keeper/querier_test.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2021 ChainSafe Systems
+// SPDX-License-Identifier: MIT
+
+package keeper
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/ChainSafe/chainlink-cosmos/x/chainlink/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func TestQuerier_GetRoundFeedData(t *testing.T) {
+	keeper, ctx := setupKeeper(t)
+	amino := codec.NewLegacyAmino()
+	querier := NewQuerier(*keeper, amino)
+	roundStore := ctx.KVStore(keeper.roundStoreKey)
+
+	testCases := []struct {
+		feedId    string
+		roundId   uint64
+		feedData  []byte
+		submitter []byte
+		insert    bool
+	}{
+		{feedId: "feed1", roundId: 100, feedData: []byte{'a', 'b', 'c'}, submitter: []byte("addressMock1"), insert: true},
+		{feedId: "feed1", roundId: 200, feedData: []byte{'d', 'e', 'f'}, submitter: []byte("addressMock2"), insert: true},
+		{feedId: "feed1", roundId: 300, feedData: []byte{'g', 'h', 'i'}, submitter: []byte("addressMock3"), insert: false},
+		{feedId: "feed2", roundId: 400, feedData: []byte{'j', 'k', 'l'}, submitter: []byte("addressMock4"), insert: false},
+	}
+
+	// Add all feed cases to store
+	for _, tc := range testCases {
+		if !tc.insert {
+			continue
+		}
+		// force set roundId-1 for SetFeedData
+		roundStore.Set(types.GetRoundIdKey(tc.feedId), i64tob(tc.roundId-1))
+
+		msgFeedData := types.MsgFeedData{
+			FeedId:    tc.feedId,
+			FeedData:  tc.feedData,
+			Submitter: tc.submitter,
+		}
+
+		_, _, err := keeper.SetFeedData(ctx, &msgFeedData)
+		require.NoError(t, err)
+	}
+
+	// Add all feed cases to store and try retrieve them
+	for _, tc := range testCases {
+		testName := fmt.Sprintf("feed:%s,round:%d", tc.feedId, tc.roundId)
+		t.Run(testName, func(t *testing.T) {
+			result, err := querier(ctx, []string{types.QueryRoundFeedData, strconv.FormatUint(tc.roundId, 10), tc.feedId}, abci.RequestQuery{})
+			require.NoError(t, err)
+
+			var roundDataResponse types.GetRoundDataResponse
+			err = amino.UnmarshalJSON(result, &roundDataResponse)
+			require.NoError(t, err)
+
+			if tc.insert {
+				require.Equal(t, 1, len(roundDataResponse.GetRoundData()))
+				require.Equal(t, tc.feedId, roundDataResponse.GetRoundData()[0].GetFeedId())
+				require.Equal(t, strconv.FormatUint(tc.roundId, 10), string(roundDataResponse.GetRoundData()[0].GetFeedData().GetContext()))
+				require.Equal(t, tc.submitter, roundDataResponse.GetRoundData()[0].GetFeedData().GetOracles())
```

Comment:
hmm.. should `submitter` be an `oracle`?

Reply:
i think wrong comparison going on here

Reply:
You right, but since OCR is not fully implemented, this is just dummy test to verify that each fields are correctly stored and retrieved

Reply:
This is how the code is currently in `keeper.go/SetFeedData`

```go
deserializedOCRReport := types.OCRAbiEncoded{
    Context:      []byte(fmt.Sprintf("%d", roundId)),
    Oracles:      feedData.Submitter.Bytes(),
    Observations: observations,
}
```

We set oracles as submitter

Reply:
with the current OCR report info, we can only know who submits the report, that's why we assigned the `oracles` to be  the submitter, but the `oracles` ideally should be an array of the cosmos account of the current round data providers including submitter itself, @araskachoi we discussed this, we will add pubkeys an all other stuff into the submit feed data tx later

Reply:
ok 👍